configure.ac

```diff
@@ -236,7 +236,14 @@ AC_CHECK_PROG(YASM_CHECK, yasm, yes)
 if test x"$YASM_CHECK" = x"yes"; then
   if yasm -f elf64 src/common/crc32c_intel_fast_asm.S -o /dev/null; then
      echo 'we have a modern and working yasm'
-     if test `arch` = "x86_64" ; then
+     # FreeBSD has uname -p not arch(1), so do not exec arch
+     arch_x86_64=0
+     if test `uname -p` = "amd64"; then
+	arch_x86_64=1
+     elif test  -x /usr/bin/arch -a `arch` = "x86_64" ; then
```

Comment:
On 22-3-2016 11:56, Kefu Chai wrote:
> In configure.ac
> <https://github.com/ceph/ceph/pull/7515#discussion_r56967193>:
> 
>> @@ -236,7 +236,14 @@ AC_CHECK_PROG(YASM_CHECK, yasm, yes)
>>  if test x"$YASM_CHECK" = x"yes"; then
>>    if yasm -f elf64 src/common/crc32c_intel_fast_asm.S -o /dev/null; then
>>       echo 'we have a modern and working yasm'
>> -     if test `arch` = "x86_64" ; then
>> +     # FreeBSD has uname -p not arch(1), so do not exec arch
>> +     arch_x86_64=0
>> +     if test `uname -p` = "amd64"; then
>> +	arch_x86_64=1
>> +     elif test  -x /usr/bin/arch -a `arch` = "x86_64" ; then
> 
> sorry, i mean, it's not consistent of how you detect the executable and
> how you call it.
> 
> probably we could put
> 
> if which arch 2>&1 > /dev/null; then
>   [ `arch` = "x86_64" ] && arch_x86_64=1
> else
>   ...

Oke, I see what you mean.
It is a bit more flexible to test with which(1) for architectures that
have arch somewhere else in their tree.
On the otherhand IF all linux have 'uname -p' it would fix the problem
for all atm.:
[wjw@centos7 ~]$ uname -p
x86_64

And run that thru a case()....